test/extended/etcd/leader_changes.go

```diff
@@ -3,20 +3,54 @@ package etcd
 import (
 	"context"
 	"fmt"
+	"strconv"
+	"strings"
 	"time"
 
 	g "github.com/onsi/ginkgo"
 	o "github.com/onsi/gomega"
 	"github.com/prometheus/common/model"
 
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 	configv1 "github.com/openshift/api/config/v1"
 	"github.com/openshift/origin/test/extended/prometheus/client"
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
+func extractMaxRevision(configMaps []corev1.ConfigMap) (int, error) {
+	var maxRevision int
+	for _, configMap := range configMaps {
+		if strings.Contains(configMap.Name, "revision-status-") {
+			sub := strings.Split(configMap.Name, "-")
+			if len(sub) != 3 {
+				fmt.Errorf("the configmap: %v has an incorrect name, unable to extract the revision number", configMap.Name)
+			}
+			revision, err := strconv.Atoi(sub[2])
+			if err != nil {
+				return 0, err
+			}
+			if revision > maxRevision {
+				maxRevision = revision
+			}
+		}
+	}
+	return maxRevision, nil
+}
+
 var _ = g.Describe("[sig-etcd] etcd", func() {
 	defer g.GinkgoRecover()
 	oc := exutil.NewCLIWithoutNamespace("etcd-leader-change").AsAdmin()
+
+	var earlyMaxRevision int
+	g.It("record the start revision of the etcd-operator [Early]", func() {
+		configMaps, err := oc.KubeClient().CoreV1().ConfigMaps("openshift-etcd").List(context.TODO(), metav1.ListOptions{})
```

Comment:
client-go has an operator client and you can read the current revision out of the etcd.operator.openshift.io status.